providers/gitlab.go

```diff
@@ -213,8 +256,37 @@ func (p *GitLabProvider) EnrichSessionState(ctx context.Context, s *sessions.Ses
 		return fmt.Errorf("group membership check failed: %v", err)
 	}
 
+	p.addProjectMembershipToState(ctx, s)
```

Comment:
Tiny bit of scope creep:

Above this line in `verifyGroupMembership` -- can you do similar logic to get it added to the state and not doing authorization logic?

Then everything can rely on the default `Authorize` functionality and we don't need to worry about two paradigms being implemented in the same provider.

Reply:
I'll do it, see https://github.com/oauth2-proxy/oauth2-proxy/pull/630#discussion_r526734384

Reply:
https://github.com/oauth2-proxy/oauth2-proxy/pull/630/commits/97564bc5d3d8f130848610ba2d4251fb40e21e0c